### PR TITLE
Notification-center - divide by feeds tabs

### PR DESCRIPTION
**Pull request by ainouzgali**
Screen Shot 2022-07-03 at 18 55 41

### CODE REVIEW

File: apps/api/src/app/widgets/widgets.controller.ts [hunk 1]
```diff
@@ -44,17 +47,34 @@ export class WidgetsController {
...
+  @Get('/notifications/feeds')
+  async getFeeds(@SubscriberSession() subscriberSession: SubscriberEntity) {
+    const command = GetFeedsCommand.create({
+      organizationId: subscriberSession._organizationId,
+      environmentId: subscriberSession._environmentId,
+      subscriberId: subscriberSession._id,
+    });
+
+    return await this.getFeedsUsecase.execute(command);
+  }
```
Comment (scopsy) [inline apps/api/src/app/widgets/widgets.controller.ts:76]: Isn't both of them duplicate endpoints? @ainouzgali

File: libs/shared/src/entities/messages/messages.interface.ts [hunk 2]
```diff
@@ -20,5 +20,13 @@ export interface IMessage {
...
   };
+  _feedId: string;
   payload: Record<string, unknown>;
 }
+
+export interface IFeedEntity {
+  _id: string;
+  name: string;
+  _environmentId: string;
+  _organizationId: string;
```
Comment (djabarovgeorge) [inline libs/shared/src/entities/messages/messages.interface.ts:31]: Due to the reason, it is related to another entity scheme, what do you think about moving it to a dedicated file?

File: apps/api/src/app/widgets/usecases/get-notifications-feed/get-notifications-feed.command.ts [hunk 3]
```diff
@@ -9,4 +9,8 @@ export class GetNotificationsFeedCommand extends EnvironmentWithSubscriber {
 
   @IsNumber()
   page: number;
+
+  @IsString()
+  @IsOptional()
+  feedId: string;
```
Comment (djabarovgeorge) [inline apps/api/src/app/widgets/usecases/get-notifications-feed/get-notifications-feed.command.ts:15]:
```suggestion
  feedId?: string;
```

File: apps/api/src/app/widgets/usecases/initialize-session/initialize-session.usecase.ts [hunk 4]
```diff
@@ -56,6 +62,7 @@ export class InitializeSession {
         lastName: subscriber.lastName,
         phone: subscriber.phone,
       },
+      feeds,
```
Comment (djabarovgeorge) [inline apps/api/src/app/widgets/usecases/initialize-session/initialize-session.usecase.ts:65]: Will all the object will be returned here?

class FeedEntity {
  _id?: string;
  name: string;
  _environmentId: string;
  _organizationId: string;
}

If so maybe we can remove all the unneeded parameters and return only feeds? what do you think?

File: libs/dal/src/repositories/message/message.entity.ts [hunk 5]
```diff
@@ -46,6 +46,8 @@ export class MessageEntity {
     };
   };
 
+  _feedId: string;
```
Comment (djabarovgeorge) [inline libs/dal/src/repositories/message/message.entity.ts:49]:
```suggestion
  _feedId?: string;
```

Is there a way the feedId will be missing? if so maybe we should allow null

File: libs/testing/src/notification-template.service.ts [hunk 6]
```diff
@@ -61,6 +66,7 @@ export class NotificationTemplateService {
         content: message.content,
         subject: message.subject,
         name: message.name,
+        _feedId: feeds[0]._id,
```
Comment (djabarovgeorge) [inline libs/testing/src/notification-template.service.ts:69]: Great job on adding support on cypress test! :)

File: packages/notification-center/src/index.ts [hunk 7]
```diff
@@ -51,4 +50,9 @@ export interface INovuProviderContext {
   subscriberHash: string;
 }
 
+export interface IFeedsContext {
+  feeds: IFeedEntity[];
```
Comment (djabarovgeorge) [inline packages/notification-center/src/index.ts:54]: So here we can save only the feed without the env params? what do you think?

File: apps/api/src/app/widgets/usecases/get-notifications-feed/get-notifications-feed.usecase.ts [hunk 8]
```diff
@@ -8,14 +8,17 @@ export class GetNotificationsFeed {
   constructor(private messageRepository: MessageRepository) {}
 
   async execute(command: GetNotificationsFeedCommand): Promise<MessageEntity[]> {
-    return await this.messageRepository.findBySubscriberChannel(
+    const result = await this.messageRepository.findBySubscriberChannel(
       command.environmentId,
       command.subscriberId,
       ChannelTypeEnum.IN_APP,
+      command.feedId,
```
Comment (djabarovgeorge) [inline apps/api/src/app/widgets/usecases/get-notifications-feed/get-notifications-feed.usecase.ts:15]: What will be returned if the user does not have any feeds?

File: packages/notification-center/src/api/api.service.ts [hunk 9]
```diff
@@ -25,8 +25,8 @@ export class ApiService {
     return await this.httpClient.post(`/widgets/messages/${messageId}/seen`, {});
   }
 
-  async getNotificationsList(page: number): Promise<IMessage[]> {
-    return await this.httpClient.get(`/widgets/notifications/feed?page=${page}`);
+  async getNotificationsList(page: number, feedId: string): Promise<IMessage[]> {
+    return await this.httpClient.get(`/widgets/notifications/feed?page=${page}&feedId=${feedId}`);
```
Comment (djabarovgeorge) [inline packages/notification-center/src/api/api.service.ts:29]: So here you may send undefined feedId, it takes me back to the question of the response with no feedId in the usecase https://github.com/novuhq/novu/pull/767/files#r913831986

File: packages/notification-center/src/components/notification-center/components/Main.tsx [hunk 10]
```diff
@@ -1,91 +1,13 @@
...
+import { FeedsTabs } from './FeedsTabs';
 
 export function Main() {
-  const { api } = useApi();
-  const { onNotificationClick, onUrlChange } = useContext(NotificationCenterContext);
-  const {
-    markAsSeen: markNotificationAsSeen,
-    fetchNextPage,
-    refetch,
-    notifications: data,
```
Comment (djabarovgeorge) [inline packages/notification-center/src/components/notification-center/components/Main.tsx:18]: Loved the refactoring part!

File: packages/notification-center/src/components/notification-center/components/Main.tsx [hunk 11]
```diff
@@ -1,91 +1,13 @@
...
-          <img src={image as any} alt="logo" style={{ maxWidth: 200 }} />
-        </div>
-      ) : (
-        <NotificationsList
-          onNotificationClicked={onNotificationClicked}
-          notifications={data || []}
-          onFetch={fetchNext}
-          hasNextPage={hasNextPage}
-        />
-      )}
```
Comment (djabarovgeorge) [inline packages/notification-center/src/components/notification-center/components/Main.tsx:76]: Loved the refactoring part!!

**Review by djabarovgeorge — COMMENTED** [review]
Great Job ✨
I left some questions about stuff I was not able to understand, and a couple of suggestions let me know what you think.

File: apps/api/src/app/widgets/usecases/get-notifications-feed/get-notifications-feed.usecase.ts (hunk 8 above)
Comment (ainouzgali) [reply apps/api/src/app/widgets/usecases/get-notifications-feed/get-notifications-feed.usecase.ts:15]: All messages, the check happens in the repository :)
Screen Shot 2022-07-06 at 12 44 39

**Comment by ainouzgali** [comment]
@djabarovgeorge Thank you so much for the detailed and thorough review! It is very appreciated!

**Pull request merged**